root/static/js/canto-modules.js

```diff
@@ -1284,7 +1284,7 @@ function openSingleGeneAddDialog($uibModal) {
 
 
 function featureChooserControlHelper($scope, $uibModal, CursGeneList,
-  CursGenotypeList, toaster) {
+  CursGenotypeList, Curs, toaster) {
```

Comment:
We have a `Metagenotype` service now that just handles metagenotypes:

https://github.com/pombase/canto/blob/72117a8a84b7e7ed3e752075d840611d3bfc9196/root/static/js/canto-modules.js#L471

It's possible you could use that instead, unless you're already aware of it and decided not to use it because you preferred the way `Curs` worked. Alternatively, we could just inject `Curs` and not use `CursGeneList` or `CursGenotypeList`, since this function and its directive might be using most of the data that `Curs` exposes  anyway. It's your choice.

(As an aside, I think `Metagenotype` should probably be renamed to `MetagenotypeService` or `CursMetagenotypeList` to make its function more obvious.)

Reply:
I didn't use the Metagenotype Service because it has quite a different interface to the other two services and it didn't fit the usage.

I like that the Metagenotype Service uses $rootScope.$broadcast() to notify about changes.  I think that's a good plan.  I didn't realise that was possible when I implemented the Curs service.

I think it would be a good clean up task to improve Curs, CursGeneList etc. to use $broadcast() and then make the interface of the Metagenotype Service match the other Services.

Reply:
> As an aside, I think Metagenotype should probably be renamed to MetagenotypeService

I'd be happy with having "Service" at the end of all services.  I haven't been very consistent.